File: projects/frontend/src/types/emailLog.ts
export const EmailStatus = {
  Pending: 0,
  Sending: 1,
  Sent: 2,
  Failed: 3,
} as const;
export type EmailStatus = (typeof EmailStatus)[keyof typeof EmailStatus];

export interface EmailLogItem {
  id: string;
  toEmail: string;
  subject: string;
  templateName: string;
  status: EmailStatus;
  retryCount: number;
  maxRetries: number;
  sentAt: string | null;
  errorMessage: string | null;
  relatedEntityType: string | null;
  relatedEntityId: string | null;
  createdDate: string;
}

File: projects/backend/src/EbayDesign.Domain/Entities/EmailLog.cs
using EbayDesign.Domain.Common;
using EbayDesign.Domain.Enums;

namespace EbayDesign.Domain.Entities;

public class EmailLog : BaseEntity
{
    public string ToEmail { get; set; } = string.Empty;
    public string Subject { get; set; } = string.Empty;
    public string TemplateName { get; set; } = string.Empty;
    public string? TemplateModel { get; set; }
    public EmailStatus Status { get; set; } = EmailStatus.Pending;
    public int RetryCount { get; set; }
    public int MaxRetries { get; set; } = 3;
    public DateTime? NextRetryAt { get; set; }
    public DateTime? SentAt { get; set; }
    public string? ErrorMessage { get; set; }
    public string? RelatedEntityType { get; set; }
    public string? RelatedEntityId { get; set; }
}
